packages/event-handler/src/rest/RouteHandlerRegistry.ts

```diff
@@ -1,10 +1,24 @@
 import type { GenericLogger } from '@aws-lambda-powertools/commons/types';
-import type { RouteRegistryOptions } from '../types/rest.js';
+import type {
+  DynamicRoute,
+  HttpMethod,
+  Path,
+  RouteHandlerOptions,
+  RouteRegistryOptions,
+} from '../types/rest.js';
+import { ParameterValidationError } from './errors.js';
 import type { Route } from './Route.js';
-import { validatePathPattern } from './utils.js';
+import {
+  compilePath,
+  processParams,
+  validateParams,
+  validatePathPattern,
+} from './utils.js';
 
 class RouteHandlerRegistry {
-  readonly #routes: Map<string, Route> = new Map();
+  readonly #staticRoutes: Map<string, Route> = new Map();
+  readonly #dynamicRoutesIndexMap: Map<string, number> = new Map();
+  readonly #dynamicRoutes: DynamicRoute[] = [];
   readonly #routesByMethod: Map<string, Route[]> = new Map();
```

Comment:
I haven't removed it here but I don't think we need this anymore.

Reply:
Feel free to remove it, it might've been a remnant from a previous pass.